MovingBlocks/Terasology
# TeraEd: Run Terasology in Swing/AWT

### Contains

Run Terasology in TeraEd

![terasology-teraed](https://user-images.githubusercontent.com/7817364/102780661-9e9a4d80-4396-11eb-9530-0cadc2103177.jpg)

### How to test

Run Terasology and check if it works unchanged (I didn't run much tests)
Run TeraEd and check if Terasology starts in it (currently only game menu works)

### Notes

This is first small step to make TeraEd do something, but I actually doubt this will be merged as it makes a bit of a (additional) mess in the code.

Changes modify rendering methods, because for the openGL to run in canvas, you need to run all opengl operations during canvas render instead of typical at any time assuming GL window exist. Also swing rendering thread needs to be used instead of main loop.

If approved, more functionality could be added to TeraEd. Feel free to recommend any improvements or if anyone know how to make it load a save/new game, let me know...
